extensions/functions/src/main/java/io/crate/operation/aggregation/HyperLogLogDistinctAggregation.java

```diff
@@ -111,13 +119,12 @@ public HllState iterate(RamAccounting ramAccounting,
                             MemoryManager memoryManager,
                             HllState state,
                             Input... args) throws CircuitBreakingException {
-        if (state.isInitialized() == false) {
-            int precision = HyperLogLogPlusPlus.DEFAULT_PRECISION;
-            if (args.length > 1) {
-                precision = DataTypes.INTEGER.sanitizeValue(args[1].value());
-            }
-            state.init(memoryManager, precision);
+
+        int precision = HyperLogLogPlusPlus.DEFAULT_PRECISION;
+        if (args.length > 1) {
+            precision = DataTypes.INTEGER.sanitizeValue(args[1].value());
         }
+        state = initIfNeeded(state, memoryManager, precision);
```

Comment:
This will trigger an additional `args[1].value()` call even if the state is already initionalized. If `args[1]` is a function or column that could be expensive. 

What is the benefit of this change?

Reply:
Wanted to have one common method for `init` and use it in 3 places. I see, semantics are different, in case of DocValueAggregators it's done in `initialState` once and for regular aggregation we can handle args only in `iterate`. Reverted initialization for regular aggregation.